src/Handler/Bucket/Share/ShareBucketHandler.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\StorageDriver\BigQuery\Handler\Bucket\Share;
+
+use Google\Cloud\BigQuery\AnalyticsHub\V1\DataExchange;
+use Google\Cloud\BigQuery\AnalyticsHub\V1\Listing;
+use Google\Cloud\BigQuery\AnalyticsHub\V1\Listing\BigQueryDatasetSource;
+use Google\Protobuf\Internal\Message;
+use Keboola\StorageDriver\BigQuery\GCPClientManager;
+use Keboola\StorageDriver\Command\Bucket\ShareBucketCommand;
+use Keboola\StorageDriver\Command\Bucket\ShareBucketResponse;
+use Keboola\StorageDriver\Contract\Driver\Command\DriverCommandHandlerInterface;
+use Keboola\StorageDriver\Credentials\GenericBackendCredentials;
+
+final class ShareBucketHandler implements DriverCommandHandlerInterface
+{
+    public GCPClientManager $clientManager;
+
+    public function __construct(GCPClientManager $clientManager)
+    {
+        $this->clientManager = $clientManager;
+    }
+
+    /**
+     * @inheritDoc
+     * @param GenericBackendCredentials $credentials
+     */
+    public function __invoke(
+        Message $credentials, // backend credentials
+        Message $command,
+        array $features
+    ): ?Message {
+        assert($credentials instanceof GenericBackendCredentials);
+        assert($command instanceof ShareBucketCommand);
+
+        assert($command->getSourceProjectId() !== '');
+        assert($command->getSourceProjectReadOnlyRoleName() !== '');
+        assert($command->getSourceBucketObjectName() !== '');
+
+        $analyticHubClient = $this->clientManager->getAnalyticHubClient($credentials);
+        $projectStringId = $command->getSourceProjectId();
+
+        $dataExchangeId = $command->getSourceProjectReadOnlyRoleName();
+
+        $formattedParent = $analyticHubClient->dataExchangeName(
+            $projectStringId,
+            GCPClientManager::DEFAULT_LOCATION,
+            $dataExchangeId
+        );
+        $listingId = $command->getSourceBucketObjectName();
+        $lst = new BigQueryDatasetSource([
+            'dataset' => sprintf(
+                'projects/%s/datasets/%s',
```

Comment:
jsem asi moc defenzivni, ale navrhoval bych tady nejakej assert na ty IDs kvuli tem lomitkum. Sice se to nastavuje v connection, ale mame cil z toho udelat polo-verejne API

Reply:
nechapem jak myslis tie aserty kvoli lomitkam ? toto je format ktory vyzaduje bigquery tie lomitka je to cesta k tomu datasetu. 
`$projectStringId` vygeneruje GCP posle nam my si to ulozime a potom to posielame, dataset je jak pises posielany z connection ako schema_name.

Naviac to validuje ten endpoint ze ked to vyplnis s nejakymi chybnymi znakmi tak ti to vynada. Ak myslis ze mame pridat asserty na to jak maju vypadat tie premenne tak to musim nejak nabusit to co podporuje GCP a myslim ze to ani tak nikde nerobim

Reply:
myslel jsem to tak, abychom checkli ,ze ty IDcka neobsahuji `/` aby to nerozbilo tu cestu . Ale jak rikam, asi to je zbytecne defenzivni

Reply:
Pridal som tam validaciu na to ze to neobsahuje `/` v postate to co sa tam posiela je vzdy v rezii bud connection alebo ineho "managera" ktory to spusta s root credentials takze musi riadit spravne to ako sa to zaklada. projektStringId vygeneruje google a ulozime si ho do db a to nasledne pouzivame to site schema name